sklearn/linear_model/_sgd_fast.pyx

```diff
@@ -500,7 +500,7 @@ def _plain_sgd(cnp.ndarray[double, ndim=1, mode='c'] weights,
     cdef double sample_weight
     cdef double class_weight = 1.0
     cdef unsigned int count = 0
-    cdef unsigned int train_count = n_samples - validation_mask.sum()
+    cdef unsigned int train_count = n_samples - np.sum(validation_mask)
     cdef unsigned int epoch = 0
```

Comment:
Once again, changing the type of `epoch` to `int` to match `max_iter`'s is preferred over using casts.

Reply:
Also, some variables defined in those lines aren't used (e.g. `w_ptr`).
Can you remove them?

Reply:
@jjerphan `epoch` and `max_iter` are supposedly unsigned. The parameter validation imposes a positive `max_iter`:

https://github.com/scikit-learn/scikit-learn/blob/1bf4ebdd1ecd89a9d9c37cc1b52bcc7c48bb36f5/sklearn/linear_model/_stochastic_gradient.py#L85

It means that we can safely cast and declare both as unsigned is closer to their real semantics.

Reply:
Ah yes, there might be some type coercion (as Python by default doesn't use `unsigned` integral types), but `unsigned int` for typing `max_iter` might be preferred. :+1:.